src/R/Components/Impl/Plots/Implementation/ViewModel/RPlotDeviceViewModel.cs

```diff
@@ -0,0 +1,279 @@
+
+            Debug.Assert(_device != null);
+            Debug.Assert(_device.ActivePlot != null);
+            await _plotManager.RemovePlotAsync(_device.ActivePlot);
+        }
+
+        public async Task ClearAllPlotsAsync() {
+            _shell.AssertIsOnMainThread();
+
+            Debug.Assert(_device != null);
+            await _plotManager.RemoveAllPlotsAsync(_device);
+        }
+
+        public async Task NextPlotAsync() {
+            _shell.AssertIsOnMainThread();
+
+            Debug.Assert(_device != null);
+            await _plotManager.NextPlotAsync(_device);
+        }
+
+        public async Task PreviousPlotAsync() {
+            _shell.AssertIsOnMainThread();
+
+            Debug.Assert(_device != null);
+            await _plotManager.PreviousPlotAsync(_device);
+        }
+
+        public async Task ActivateDeviceAsync() {
+            _shell.AssertIsOnMainThread();
+
+            if (_device == null) {
+                await _plotManager.NewDeviceAsync(InstanceId);
+            } else {
+                await _plotManager.ActivateDeviceAsync(_device);
+            }
+        }
+
+        public void ClickPlot(int pixelX, int pixelY) {
+            _shell.AssertIsOnMainThread();
+
+            if (LocatorMode) {
+                var result = LocatorResult.CreateClicked(pixelX, pixelY);
+                EndLocatorMode(result);
+            }
+        }
+
+        public async Task CopyPlotFromAsync(Guid sourceDeviceId, Guid sourcePlotId, bool isMove) {
+            _shell.AssertIsOnMainThread();
+
+            if (_device == null) {
+                await _plotManager.NewDeviceAsync(InstanceId);
+            }
+
+            Debug.Assert(_device != null);
+            await _plotManager.CopyOrMovePlotFromAsync(sourceDeviceId, sourcePlotId, _device, isMove);
+        }
+
+        public async Task<LocatorResult> StartLocatorModeAsync(CancellationToken ct) {
+            _shell.AssertIsOnMainThread();
+
+            _locatorTcs = new TaskCompletionSource<LocatorResult>();
+            ct.Register(EndLocatorMode);
+
+            LocatorMode = true;
+            LocatorModeChanged?.Invoke(this, EventArgs.Empty);
+
+            _device.LocatorMode = LocatorMode;
+
+            var task = _locatorTcs.Task;
+            return await task;
+        }
+
+        public void EndLocatorMode() {
+            _shell.AssertIsOnMainThread();
+
+            EndLocatorMode(LocatorResult.CreateNotClicked());
+        }
+
+        public void EndLocatorMode(LocatorResult result) {
+            _shell.AssertIsOnMainThread();
+
+            var tcs = _locatorTcs;
+            _locatorTcs = null;
+            tcs?.SetResult(result);
+
+            LocatorMode = false;
+            LocatorModeChanged?.Invoke(this, EventArgs.Empty);
+
+            _device.LocatorMode = LocatorMode;
+        }
+
+        private void DeviceNumChanged(object sender, RPlotDeviceEventArgs e) {
+            DeviceNameChanged?.Invoke(this, EventArgs.Empty);
```

Comment:
Ultimately, it's so that the visual component can call on its container to update the tool window caption.